community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/LogicalPlanConverter.scala

```diff
@@ -203,6 +203,10 @@ object LogicalPlanConverter {
           s"Failed trying to rewrite $classNameV3_3 - 3.4 class not found ($classNameV3_4)", e)
         case Failure(e: NoSuchElementException) => throw new InternalException(
           s"Failed trying to rewrite $classNameV3_3 - this class does not have a constructor", e)
+        case Failure(e: Exception) => throw new InternalException(
+          s"Failed trying to rewrite $classNameV3_3", e)
+        case Failure(e: Throwable) => throw new InternalException(
```

Comment:
I am not sure it makes sense to catch all Throwables. A Throwable that is not an Exception is usually something very bad like OutOfMemoryError, which you wouldn't like to hide in another Exception.